syntax = "proto3";

package SpotPlusArm;

message dexterConfiguration{

  message Joint{
    double currentAngle = 1;
    double goalAngle = 2;
  }

  Joint j1 = 1;
  Joint j2 = 2;
  Joint j3 = 3;
  Joint j4 = 4;
  Joint j5 = 5;
}

enum MoveType{
    P_MOVE = 0;
    A_MOVE = 1;
}

message dexterRequest5{
  double j1 = 1;
  double j2 = 2;
  double j3 = 3;
  double j4 = 4;
  double j5 = 5;
  MoveType commandType = 6;
}

message dexterRequest7{
  double j1 = 1;
  double j2 = 2;
  double j3 = 3;
  double j4 = 4;
  double j5 = 5;
  double j6 = 6;
  double j7 = 7;
  MoveType commandType = 8;
}

//
message dexterStatus{
  int32 status = 1;
}
